LoopMusicViewController.m

```diff
@@ -893,13 +886,31 @@ - (NSArray*)getSongIndices
 {
     /// The IDs of all tracks in the current playlist.
     NSArray *splitSongs = nil;
-    [self prepareQuery:[NSString stringWithFormat:@"SELECT tracks FROM Playlists WHERE id = %ld", (long)playlistIndex]];
-    if (sqlite3_step(statement) == SQLITE_ROW)
+    NSString *trackString = nil;
+    if(playlistIndex)
     {
-        /// The database string containing the current playlist's track IDs.
-        NSString *trackString = [[NSString alloc] initWithUTF8String:(const char *) sqlite3_column_text(statement, 0)];
-        splitSongs = [trackString componentsSeparatedByString:@","];
+        [self prepareQuery:[NSString stringWithFormat:@"SELECT tracks FROM Playlists WHERE id = %ld", (long)playlistIndex]];
+        
+        if (sqlite3_step(statement) == SQLITE_ROW)
+        {
+            /// The database string containing the current playlist's track IDs.
+            trackString = [[NSString alloc] initWithUTF8String:(const char *) sqlite3_column_text(statement, 0)];
+        }
     }
+    else
+    {
+        [self prepareQuery:[NSString stringWithFormat:@"SELECT id FROM Tracks"]];
+        trackString = @"";
+        while (sqlite3_step(statement) == SQLITE_ROW)
+        {
+            /// The database string containing the current track ID
+            NSString *idString = [[NSString alloc] initWithUTF8String:(const char *) sqlite3_column_text(statement, 0)];
+            trackString = [[trackString stringByAppendingString:idString] stringByAppendingString:@","];
```

Comment:
This concatenates all IDs together as a string before splitting that string back up again. Can the IDs be immediately added to an array as they are processed?

Reply:
I couldn't figure out how to do it... As far as I can tell you can't initialize a blank NSArray, and a mutable array doesn't work here because of the function return value. Suggestions?

Reply:
The copy function can return an immutable NSArray from a NSMutableArray.